packages/fx-core/src/common/spec-parser/utils.ts

```diff
@@ -129,16 +129,27 @@ export function isSupportedApi(
   method: string,
   path: string,
   spec: OpenAPIV3.Document,
-  allowMissingId: boolean
+  allowMissingId: boolean,
+  allowAPIKeyAuth: boolean
 ): boolean {
   const pathObj = spec.paths[path];
   method = method.toLocaleLowerCase();
   if (pathObj) {
     if (
       (method === ConstantString.PostMethod || method === ConstantString.GetMethod) &&
-      pathObj[method] &&
-      !pathObj[method]?.security
+      pathObj[method]
     ) {
+      const securities = pathObj[method]!.security;
+      const apiKeyAuth = getAPIKeyAuth(securities, spec);
+
+      if (!allowAPIKeyAuth && securities) {
+        return false;
+      }
+
+      if (allowAPIKeyAuth && securities && !apiKeyAuth) {
+        return false;
```

Comment:
If the api key and logic will still be shown?

Reply:
This condition will filter other auth such as aad for current stage